tests/integration/layers/layersManager.spec.ts

```diff
@@ -38,4 +44,196 @@ describe('layerManager', function () {
+      expect(response.body).toEqual({ message: conflictErrorMessage });
+    });
+  });
+
+  describe('#updateLayer', function () {
+    const mockUpdateLayerRequest: ILayerPostRequest = {
+      name: 'amsterdam_5cm',
+      tilesPath: '/path/to/tiles/directory/in/my/bucket/',
+      maxZoomLevel: 18,
+      description: 'description for amsterdam layer',
+    };
+
+    it('Happy Path', async function () {
+      const response = await requestSender.updateLayer(mockLayerNameAlreadyExists.name, mockUpdateLayerRequest);
+
+      expect(response.status).toBe(httpStatusCodes.ACCEPTED);
+    });
+
+    it('Bad Path', async function () {
+      const mockBadRequest = ({
+        mockName: 'amsterdam_5cm',
+        tilesPath: '/path/to/tiles/directory/in/my/bucket/',
+        maxZoomLevel: 18,
+        description: 'description for amsterdam layer',
+      } as unknown) as ILayerPostRequest;
+
+      const response = await requestSender.updateLayer(mockLayerNameAlreadyExists.name, mockBadRequest);
+
+      expect(response.status).toBe(httpStatusCodes.BAD_REQUEST);
+    });
+
+    it('Sad Path', async function () {
+      const response = await requestSender.updateLayer(mockLayerNameIsNotExists.name, mockUpdateLayerRequest);
+      const notFoundErrorMessage = `Layer name '${mockLayerNameIsNotExists.name}' is not exists`;
+
+      expect(response.status).toBe(httpStatusCodes.NOT_FOUND);
+      expect(response.body).toEqual({ message: notFoundErrorMessage });
+    });
+  });
+
+  describe('#removeLayer', function () {
+    it('Happy Path', async function () {
+      const response = await requestSender.removeLayer(mockLayerNameAlreadyExists.name);
+
+      expect(response.status).toBe(httpStatusCodes.ACCEPTED);
+    });
+
+    it('Sad Path', async function () {
+      const response = await requestSender.removeLayer(mockLayerNameIsNotExists.name);
+      const notFoundErrorMessage = `Layer name '${mockLayerNameIsNotExists.name}' is not exists`;
+
+      expect(response.status).toBe(httpStatusCodes.NOT_FOUND);
+      expect(response.body).toEqual({ message: notFoundErrorMessage });
+    });
+  });
+
+  describe('#addLayerToMosaic', function () {
+    const mockMosaicName = 'existsMosaicName';
+    const mockLayerToMosaicRequest: ILayerToMosaicRequest = {
+      layerName: 'mockLayerNameExists',
+    };
+
+    it('Happy Path', async function () {
+      const response = await requestSender.addLayerToMosaic(mockMosaicName, mockLayerToMosaicRequest);
+
+      expect(response.status).toBe(httpStatusCodes.CREATED);
+    });
+
+    it('Bad Path', async function () {
+      const mockMosaicName = 'mosaicMockName';
+      const mockBadRequestRequest = ({
+        mockName: 'layerNameIsNotExists',
+      } as unknown) as ILayerToMosaicRequest;
+      const response = await requestSender.addLayerToMosaic(mockMosaicName, mockBadRequestRequest);
+
+      expect(response.status).toBe(httpStatusCodes.BAD_REQUEST);
+    });
+
+    it('Sad Path', async function () {
```

Comment:
explain 
"Sad Path - mosaic not exist"
"Sad Path - layer not exist"
etc.

Reply:
fixed in 2928d1e